src/ui/screen_saver.c

```diff
@@ -74,3 +75,13 @@ void screen_saver_reset(void)
     }
     _counter = 0;
 }
+
+void screen_saver_disable(void)
+{
+    _disabled = true;
```

Comment:
shouldn't the counter be reset here? would be strange to exit e.g. a confirmiation dialog straight into the screensaver.